File: mobile/ios/TeleSol/Models/SensorData.swift
// ════════════════════════════════════════════════════════════════
// TeleSol iOS — Data Models
// ════════════════════════════════════════════════════════════════

import Foundation

// MARK: - Outbound: Phone → Backend

struct MobileRegistration: Codable {
    let deviceId: String
    let deviceName: String
    let osType: String
    let osVersion: String
    let appVersion: String

    enum CodingKeys: String, CodingKey {
        case deviceId = "device_id"
        case deviceName = "device_name"
        case osType = "os_type"
        case osVersion = "os_version"
        case appVersion = "app_version"
    }
}

struct MobileStreamPayload: Codable {
    let deviceId: String
    let timestamp: Int64
    let noiseDb: Double
    let peakAmplitude: Double
    let isCrowdNoise: Bool
    let personCount: Int
    let motionDetected: Bool
    let gpsLatitude: Double?
    let gpsLongitude: Double?
    let gpsAccuracy: Double?
    let batteryLevel: Int?

    enum CodingKeys: String, CodingKey {
        case deviceId = "device_id"
        case timestamp
        case noiseDb = "noise_db"
        case peakAmplitude = "peak_amplitude"
        case isCrowdNoise = "is_crowd_noise"
        case personCount = "person_count"
        case motionDetected = "motion_detected"
        case gpsLatitude = "gps_latitude"
        case gpsLongitude = "gps_longitude"
        case gpsAccuracy = "gps_accuracy"
        case batteryLevel = "battery_level"
    }
}

// MARK: - Inbound: Backend → Phone

struct StreamResponse: Codable {
    let status: String
    let deviceId: String
    let fusedCrs: Double
    let fusedLevel: String
    let espConnected: Bool
    let timestamp: String

    enum CodingKeys: String, CodingKey {
        case status
        case deviceId = "device_id"
        case fusedCrs = "fused_crs"
        case fusedLevel = "fused_level"
        case espConnected = "esp_connected"
        case timestamp
    }
}

struct RegisterResponse: Codable {
    let status: String
    let deviceId: String
    let message: String
    let backendVersion: String
    let timestamp: String

    enum CodingKeys: String, CodingKey {
        case status
        case deviceId = "device_id"
        case message
        case backendVersion = "backend_version"
        case timestamp
    }
}

// MARK: - Dashboard Data (WebSocket)

struct DashboardData: Codable {
    let stats: DashboardStats?
    let liveData: LiveSensorData?
    let crs: CRSData?

    enum CodingKeys: String, CodingKey {
        case stats
        case liveData
        case crs
    }
}

struct DashboardStats: Codable {
    let totalNodes: Int?
    let activeNodes: Int?
    let activeDevices: Int?
    let riskScore: Double?
    let lastUpdate: String?
    let phoneConnected: Bool?
}

struct LiveSensorData: Codable {
    let radarHumans: Int?
    let radarDistance: Int?
    let radarEnergy: Int?
    let noiseLevel: Double?
    let tofDistance: Int?
    let vibration: Double?
    let temperature: Double?
    let cameraPeopleCount: Int?
    let crowdDensity: Double?
}

struct CRSData: Codable {
    let crs: Double?
    let level: String?
}

File: mobile/android/TeleSol/app/src/main/java/com/telesol/companion/data/models/SensorData.kt
// ════════════════════════════════════════════════════════════════
// TeleSol Android — Data Models
// ════════════════════════════════════════════════════════════════

package com.telesol.companion.data.models

import com.google.gson.annotations.SerializedName

// ── Outbound: Phone → Backend ───────────────────────────────

data class MobileRegistration(
    @SerializedName("device_id") val deviceId: String,
    @SerializedName("device_name") val deviceName: String,
    @SerializedName("os_type") val osType: String = "android",
    @SerializedName("os_version") val osVersion: String,
    @SerializedName("app_version") val appVersion: String = "1.0.0"
)

data class MobileStreamPayload(
    @SerializedName("device_id") val deviceId: String,
    val timestamp: Long,
    @SerializedName("noise_db") val noiseDb: Double = 0.0,
    @SerializedName("peak_amplitude") val peakAmplitude: Double = 0.0,
    @SerializedName("is_crowd_noise") val isCrowdNoise: Boolean = false,
    @SerializedName("person_count") val personCount: Int = 0,
    @SerializedName("motion_detected") val motionDetected: Boolean = false,
    @SerializedName("gps_latitude") val gpsLatitude: Double? = null,
    @SerializedName("gps_longitude") val gpsLongitude: Double? = null,
    @SerializedName("gps_accuracy") val gpsAccuracy: Double? = null,
    @SerializedName("battery_level") val batteryLevel: Int? = null
)

// ── Inbound: Backend → Phone ────────────────────────────────

data class StreamResponse(
    val status: String,
    @SerializedName("device_id") val deviceId: String,
    @SerializedName("fused_crs") val fusedCrs: Double,
    @SerializedName("fused_level") val fusedLevel: String,
    @SerializedName("esp_connected") val espConnected: Boolean,
    val timestamp: String
)

data class RegisterResponse(
    val status: String,
    @SerializedName("device_id") val deviceId: String,
    val message: String,
    @SerializedName("backend_version") val backendVersion: String,
    val timestamp: String
)

// ── Dashboard Data ──────────────────────────────────────────

data class DashboardData(
    val stats: DashboardStats?,
    val liveData: LiveSensorData?,
    val crs: CRSData?
)

data class DashboardStats(
    val totalNodes: Int?,
    val activeNodes: Int?,
    val activeDevices: Int?,
    val riskScore: Double?,
    val lastUpdate: String?,
    val phoneConnected: Boolean?
)

data class LiveSensorData(
    val radarHumans: Int?,
    val radarDistance: Int?,
    val radarEnergy: Int?,
    val noiseLevel: Double?,
    val tofDistance: Int?,
    val vibration: Double?,
    val temperature: Double?,
    val cameraPeopleCount: Int?,
    val crowdDensity: Double?
)

data class CRSData(
    val crs: Double?,
    val level: String?
)
